homeassistant/components/switch/rest.py

```diff
@@ -69,8 +69,9 @@ def is_on(self):
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
+        headers = {'content-type': 'application/x-www-form-urlencoded'}
```

Comment:
Thanks for guiding me, I think I've managed to make it work. 
Please let me know if there is a better way to do it. 

Thanks again!